### PR TITLE
DD-5: Direct Debit Mandate Custom Fields

### PR DESCRIPTION
**Pull request by AWG-Bohdan**
In this PR, the skeleton extension for manualdirectdebit extension is created, and the list of custom fields mentioned in DD-5 ticket are added via XML.

### CODE REVIEW

**Comment by AWG-Bohdan** [comment]
Questions:
1. Should we create the FK constraints for table fields? See comments in sql/install.sql file

File: CRM/ManualDirectDebit/Upgrader.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+/**
+ * Collection of upgrade steps.
+ */
+class CRM_ManualDirectDebit_Upgrader extends CRM_ManualDirectDebit_Upgrader_Base {
+
+  // By convention, functions that look like "function upgrade_NNNN()" are
+  // upgrade tasks. They are executed in order (like Drupal's hook_update_N).
+
+  /**
+   * Example: Run an external SQL script when the module is installed.
```
Comment (omarabuhussein) [inline CRM/ManualDirectDebit/Upgrader.php]: please remove these auto-generated docblocks from all this upgrader class methods.

File: sql/uninstall.sql [hunk 2]
```diff
@@ -0,0 +1,7 @@
+-- /*******************************************************
+-- * Delete direct debit tables
+-- *******************************************************/
+
+--SET FOREIGN_KEY_CHECKS=0;
+
+DROP TABLE IF EXISTS `dd_contribution_recurr_mandate_ref`;
```
Comment (omarabuhussein) [inline sql/uninstall.sql]: please add space at the end of the file

File: sql/install.sql [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+CREATE TABLE IF NOT EXISTS `dd_contribution_recurr_mandate_ref` (
+  `id` INT unsigned NOT NULL AUTO_INCREMENT COMMENT 'ID',
+  `recurr_id` INT(11) NULL COMMENT 'FK to Recurrent ID ....',
+  `mandate_id` INT(11) NULL COMMENT 'FK to Mandate ID ....',
+   PRIMARY KEY (`id`)
+--   CONSTRAINT FK_....._recurr_id FOREIGN KEY (`recurr_id`) REFERENCES `.....`(`id`) ON DELETE SET NULL,
+--   CONSTRAINT FK_....._mandate_id FOREIGN KEY (`mandate_id`) REFERENCES `....`(`id`) ON DELETE SET NULL,
+);
+-- TODO: Add table names for FK
```
Comment (omarabuhussein) [inline sql/install.sql]: please add space at the end of the file here too

File: manualdirectdebit.php [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+}
+
+// --- Functions below this ship commented out. Uncomment as required. ---
+
+/**
+ * Implements hook_civicrm_preProcess().
+ *
+ * @link http://wiki.civicrm.org/confluence/display/CRMDOC/hook_civicrm_preProcess
+ *
+function manualdirectdebit_civicrm_preProcess($formName, &$form) {
```
Comment (omarabuhussein) [inline manualdirectdebit.php]: please remove these commented out functions (manualdirectdebit_civicrm_preProcess and manualdirectdebit_civicrm_navigationMenu)

File: CRM/ManualDirectDebit/Upgrader.php (hunk 1 above)
Comment (AWG-Bohdan) [reply CRM/ManualDirectDebit/Upgrader.php]: don

File: sql/uninstall.sql (hunk 2 above)
Comment (AWG-Bohdan) [reply sql/uninstall.sql]: ok

File: sql/install.sql (hunk 3 above)
Comment (AWG-Bohdan) [reply sql/install.sql]: ok

File: manualdirectdebit.php (hunk 4 above)
Comment (AWG-Bohdan) [reply manualdirectdebit.php]: ok

**Comment by AWG-Bohdan** [comment]
4- The PR is supposed to add some custom fields/groups but I cannot see that you are adding any.

For this extension we create the custom fields by using XML file - xml\customFields_install.xml
>>"All fields in the table below should be mapped using XML in the extension."

**Comment by omarabuhussein** [comment]
@AWG-Bohdan 

Most things are fine now, but please check point 2, 3 and 6 in the initital review : 
https://github.com/compucorp/uk.co.compucorp.manualdirectdebit/pull/1#pullrequestreview-98239622

for point 2, you need to use interactive rebase with reword option to add DD-5 .

for point 3, please do rough documentation in the PR description and remove "**ready for review**", there is no need to write the custom fields added, something like this would do it for this PR : 

```
In this PR, the skeleton extension for manualdirectdebit extension is created, and the list of custom fields mentioned in DD-5 ticket are added via XML.

Also a new table **dd_contribution_recurr_mandate_ref** is added, "describe its purpose here" ..etc
```


for point 6, it is about dd_contribution_recurr_mandate_ref, I cannot see it mentioned in DD-5 ticket and I am not sure what it is about, can you elaborate

**Comment by AWG-Bohdan** [comment]
Done

**Comment by omarabuhussein** [comment]
approved @AWG-Bohdan

**Pull request merged**